src/peer/node.js

```diff
@@ -0,0 +1,82 @@
+const libp2p = require('libp2p-ipfs')
+const PeerId = require('peer-id')
+const PeerInfo = require('peer-info')
+const Multiaddr = require('multiaddr')
+const b58 = require('bs58')
+const pb = require('../protobuf')
+const pull = require('pull-stream')
+const { protobufStream } = require('./util')
+const binstring = require('binstring')
+
+const DEFAULT_LISTEN_ADDR = Multiaddr('/ip4/127.0.0.1/tcp/9002')
+
+class MediachainNode extends libp2p.Node {
+  directory: PeerInfo
+
+  constructor (peerId: PeerId, dirInfo: PeerInfo, listenAddrs: Array<Multiaddr> = [DEFAULT_LISTEN_ADDR]) {
+    const peerInfo = new PeerInfo(peerId)
+    listenAddrs.forEach((addr: Multiaddr) => {
+      peerInfo.multiaddr.add(addr)
+    })
+
+    super(peerInfo)
+    this.directory = dirInfo
+  }
+
+  lookup (peerId: string): Promise<PeerInfo> {
+    let mhash: Buffer
+    try {
+      mhash = new Buffer(b58.decode(peerId), 'binary')
+    } catch (err) {
+      return Promise.reject(err)
+    }
+
+    const req = pb.dir.LookupPeerRequest.encode({
+      id: binstring(mhash, {in: 'buffer', out: 'binary'})
+    })
+
+    return new Promise((resolve, reject) => {
+      this.dialByPeerInfo(this.directory, '/mediachain/dir/lookup', (err: ?Error, conn: any) => { // TODO: type for conn
```

Comment:
For example, would be nice to rewrite the `Node` class with a `Promise` based interface to avoid this awkward calling pattern